test/ConstExtraction/ExtractResultBuilders.swift

```diff
@@ -0,0 +1,108 @@
+    }
+}
+
+public struct MyFooProviderInferred: FooProvider {
+    public static var foos: [Foo] {
+        Foo(name: "CCC")
+        Foo(name: "DDD")
+    }
+}
+
+// CHECK: [
+// CHECK-NEXT:    {
+// CHECK-NEXT:      "typeName": "ExtractResultBuilders.MyFooProvider",
+// CHECK-NEXT:      "kind": "struct",
+// CHECK-NEXT:      "file": "{{.*}}test{{/|\\\\}}ConstExtraction{{/|\\\\}}ExtractResultBuilders.swift",
+// CHECK-NEXT:      "line": 28,
+// CHECK-NEXT:      "properties": [
+// CHECK-NEXT:        {
+// CHECK-NEXT:          "label": "foos",
+// CHECK-NEXT:          "type": "[ExtractResultBuilders.Foo]",
+// CHECK-NEXT:          "isStatic": "true",
+// CHECK-NEXT:          "isComputed": "true",
+// CHECK-NEXT:          "file": "{{.*}}test{{/|\\\\}}ConstExtraction{{/|\\\\}}ExtractResultBuilders.swift",
+// CHECK-NEXT:          "line": 30,
+// CHECK-NEXT:          "valueKind": "Runtime",
+// CHECK-NEXT:          "resultBuilder": {
+// CHECK-NEXT:            "type": "ExtractResultBuilders.FooBuilder",
+// CHECK-NEXT:            "file": "{{.*}}test{{/|\\\\}}ConstExtraction{{/|\\\\}}ExtractResultBuilders.swift",
+// CHECK-NEXT:            "line": 29
+// CHECK-NEXT:          }
+// CHECK-NEXT:        },
+// CHECK-NEXT:        {
+// CHECK-NEXT:          "label": "fooTwo",
+// CHECK-NEXT:          "type": "[ExtractResultBuilders.Foo]",
+// CHECK-NEXT:          "isStatic": "true",
+// CHECK-NEXT:          "isComputed": "true",
+// CHECK-NEXT:          "file": "{{.*}}test{{/|\\\\}}ConstExtraction{{/|\\\\}}ExtractResultBuilders.swift",
+// CHECK-NEXT:          "line": 36,
+// CHECK-NEXT:          "valueKind": "Runtime",
+// CHECK-NEXT:          "resultBuilder": {
+// CHECK-NEXT:            "type": "ExtractResultBuilders.FooBuilder",
+// CHECK-NEXT:            "file": "{{.*}}test{{/|\\\\}}ConstExtraction{{/|\\\\}}ExtractResultBuilders.swift",
+// CHECK-NEXT:            "line": 35
+// CHECK-NEXT:          }
+// CHECK-NEXT:        }
+// CHECK-NEXT:      ]
+// CHECK-NEXT:    },
+// CHECK-NEXT:    {
+// CHECK-NEXT:      "typeName": "ExtractResultBuilders.MyFooProviderInferred",
+// CHECK-NEXT:      "kind": "struct",
+// CHECK-NEXT:      "file": "{{.*}}test{{/|\\\\}}ConstExtraction{{/|\\\\}}ExtractResultBuilders.swift",
+// CHECK-NEXT:      "line": 42,
+// CHECK-NEXT:      "properties": [
+// CHECK-NEXT:        {
+// CHECK-NEXT:          "label": "foos",
+// CHECK-NEXT:          "type": "[ExtractResultBuilders.Foo]",
+// CHECK-NEXT:          "isStatic": "true",
+// CHECK-NEXT:          "isComputed": "true",
+// CHECK-NEXT:          "file": "{{.*}}test{{/|\\\\}}ConstExtraction{{/|\\\\}}ExtractResultBuilders.swift",
+// CHECK-NEXT:          "line": 43,
+// CHECK-NEXT:          "valueKind": "Runtime",
+// CHECK-NEXT:          "resultBuilder": {
+// CHECK-NEXT:            "type": "ExtractResultBuilders.FooBuilder",
+// CHECK-NEXT:            "file": "{{.*}}test{{/|\\\\}}ConstExtraction{{/|\\\\}}ExtractResultBuilders.swift",
+// CHECK-NEXT:            "line": 24
```

Comment:
For inferred conformances, this is where the protocol is defined. Do we maybe not need file/line information since we have the var file/line above?

Reply:
My intuition is that we don't need file/line for the attribute at all, those are largely mostly relevant for the property decl itself. But perhaps I'm missing some use-case?

Reply:
Agreed. Removing.